pkg/util/ovs/fake_ovs.go

```diff
@@ -0,0 +1,447 @@
+package ovs
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// ovsFake implements a fake ovs.Interface for testing purposes
+type ovsFake struct {
+	bridge string
+
+	ports map[string]int
+	flows []ovsFlow
+}
+
+// ovsFlow represents an OVS flow
+type ovsFlow struct {
+	table    int
+	priority int
+	created  time.Time
+	cookie   string
+	fields   []ovsField
+	actions  string
+}
+
+type ovsField struct {
+	name  string
+	value string
+}
+
+const (
+	minPriority     = 0
+	defaultPriority = 32768
+	maxPriority     = 65535
+)
+
+// NewFake returns a new ovs.Interface
+func NewFake(bridge string) Interface {
+	return &ovsFake{bridge: bridge}
+}
+
+func (fake *ovsFake) AddBridge(properties ...string) error {
```

Comment:
The bridge name is part of the ovs.Interface object; it's an implicit argument to *all* of the methods. (eg, "AddFlows" adds flows to a specific bridge, but does not have a bridge name argument).

Note that although the ovsFake implementation of AddBridge looks more like "set up bridge" than "add bridge", it does semantically behave as "AddBridge". Eg, if you do

    ovsif := ovs.NewFake("br0")
    _, err := ovsif.AddPort("veth1234", -1)

then you'll get back an error "no bridge named br0", because you didn't call AddBridge().